kernel/src/sched/priority_scheduler.rs

```diff
@@ -235,15 +236,25 @@ impl PreemptSchedInfo for Task {
     const REAL_TIME_TASK_PRIORITY: Self::PRIORITY = Priority::new(100);
 
     fn priority(&self) -> Self::PRIORITY {
-        self.schedule_info().priority
+        self.data().downcast_ref::<Weak<Thread>>().map_or_else(
+            || {
+                panic!("Found a Task without any Thread binding to it");
+            },
+            |weak_thread| weak_thread.upgrade().unwrap().priority(),
+        )
     }
 
     fn cpu(&self) -> &AtomicCpuId {
         &self.schedule_info().cpu
     }
 
-    fn cpu_affinity(&self) -> &CpuSet {
-        &self.schedule_info().cpu_affinity
+    fn cpu_affinity(&self) -> CpuSet {
+        self.data().downcast_ref::<Weak<Thread>>().map_or_else(
+            || {
+                panic!("Found a Task without any Thread binding to it");
+            },
+            |weak_thread| weak_thread.upgrade().unwrap().cpu_affinity(),
```

Comment:
I don't have any better solution (yet). But I guess that this is going to lead to performance problem. Could you think of a more efficient means to get task-associated scheduling objects?

Reply:
I've been thinking about changing  `aster-nix`'s minimal schedulable unit to `aster-nix::thread::Thread`. 

Logically, `Task` is OSTD's execution entity exposed to its users. Kernel developers may want to add some fields to `Task` to meet their needs therefore we have a `Thread` in `aster-nix`. Actually this `Thread` struct is like the `task_struct` in `aster-nix`. We should fetch the information we want directly from this struct in `aster-nix`.

I haven't try out yet. But I think what we need to do is basically two things:
1. change the OSTD `SCHEDULER` definition and relevant parts
    ```rust
    static SCHEDULER: Once<&'static dyn Scheduler<dyn TaskWrapper>> = Once::new();
    
    trait TaskWrapper {
        fn inner_task() -> &Arc<Task>;
    }
    ```
2. modify `aster-nix`'s scheduler implementation to accept a trait object (`dyn TaskWrapper`) while enqueueing.
The downcast is still not avoidable when we enqueue a task. However, scheduler-internal access to information like `priority` and `cpu_affinity` no longer need the downcast. This advantage can be notable when we implement more complicated policies (e.g., CFS involves much vruntime computation using `priority`).